OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -233,17 +233,27 @@ public Aircraft(ActorInitializer init, AircraftInfo info)
 				SetPosition(self, init.Get<CenterPositionInit, WPos>());
 
 			Facing = init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : Info.InitialFacing;
-			LandAltitude = info.LandAltitude;
 		}
 
-		public void AddLandingOffset(int offset)
+		public WDist LandAltitude
 		{
-			LandAltitude += new WDist(offset);
+			get
+			{
+				var alt = Info.LandAltitude;
+				foreach (var offset in self.TraitsImplementing<IPositionOffset>())
```

Comment:
Wouldnt it make sense to query this in the constructor?

Reply:
The value returned by `PositionOffset()` does not remain constant during the lifetime of the actor. In the case of `Carryall.PositionOffset` it changes depending on what unit it is carrying and even the facing at that particular moment.

Reply:
I meant the query for getting the TraitsImplementing<IPositionOffset>() :smile:

Reply:
Fixed.